// @ariakit/react-core/cjs/__chunks/72JGZWYP.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _2P3WACMLcjs = require('./2P3WACML.cjs');


var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');

// src/tag/tag-context.tsx
var _react = require('react');
var TagValueContext = _react.createContext.call(void 0, null);
var TagRemoveIdContext = _react.createContext.call(void 0, 
  null
);
var ctx = _MFAH5VLWcjs.createStoreContext.call(void 0, 
  [_2P3WACMLcjs.CompositeContextProvider],
  [_2P3WACMLcjs.CompositeScopedContextProvider]
);
var useTagContext = ctx.useContext;
var useTagScopedContext = ctx.useScopedContext;
var useTagProviderContext = ctx.useProviderContext;
var TagContextProvider = ctx.ContextProvider;
var TagScopedContextProvider = ctx.ScopedContextProvider;









exports.TagValueContext = TagValueContext; exports.TagRemoveIdContext = TagRemoveIdContext; exports.useTagContext = useTagContext; exports.useTagScopedContext = useTagScopedContext; exports.useTagProviderContext = useTagProviderContext; exports.TagContextProvider = TagContextProvider; exports.TagScopedContextProvider = TagScopedContextProvider;
