// @ariakit/react-core/cjs/__chunks/BB26STNR.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _2P3WACMLcjs = require('./2P3WACML.cjs');



var _5X5VM34Kcjs = require('./5X5VM34K.cjs');


var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');

// src/combobox/combobox-context.tsx
var _react = require('react');
var ComboboxListRoleContext = _react.createContext.call(void 0, 
  void 0
);
var ctx = _MFAH5VLWcjs.createStoreContext.call(void 0, 
  [_5X5VM34Kcjs.PopoverContextProvider, _2P3WACMLcjs.CompositeContextProvider],
  [_5X5VM34Kcjs.PopoverScopedContextProvider, _2P3WACMLcjs.CompositeScopedContextProvider]
);
var useComboboxContext = ctx.useContext;
var useComboboxScopedContext = ctx.useScopedContext;
var useComboboxProviderContext = ctx.useProviderContext;
var ComboboxContextProvider = ctx.ContextProvider;
var ComboboxScopedContextProvider = ctx.ScopedContextProvider;
var ComboboxItemValueContext = _react.createContext.call(void 0, 
  void 0
);
var ComboboxItemCheckedContext = _react.createContext.call(void 0, false);










exports.ComboboxListRoleContext = ComboboxListRoleContext; exports.useComboboxContext = useComboboxContext; exports.useComboboxScopedContext = useComboboxScopedContext; exports.useComboboxProviderContext = useComboboxProviderContext; exports.ComboboxContextProvider = ComboboxContextProvider; exports.ComboboxScopedContextProvider = ComboboxScopedContextProvider; exports.ComboboxItemValueContext = ComboboxItemValueContext; exports.ComboboxItemCheckedContext = ComboboxItemCheckedContext;
