// @ariakit/react-core/cjs/__chunks/CSAYKKWD.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});



var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');



var _7EQBAZ46cjs = require('./7EQBAZ46.cjs');

// src/visually-hidden/visually-hidden.tsx
var TagName = "span";
var useVisuallyHidden = _MFAH5VLWcjs.createHook.call(void 0, 
  function useVisuallyHidden2(props) {
    props = _7EQBAZ46cjs.__spreadProps.call(void 0, _7EQBAZ46cjs.__spreadValues.call(void 0, {}, props), {
      style: _7EQBAZ46cjs.__spreadValues.call(void 0, {
        border: 0,
        clip: "rect(0 0 0 0)",
        height: "1px",
        margin: "-1px",
        overflow: "hidden",
        padding: 0,
        position: "absolute",
        whiteSpace: "nowrap",
        width: "1px"
      }, props.style)
    });
    return props;
  }
);
var VisuallyHidden = _MFAH5VLWcjs.forwardRef.call(void 0, function VisuallyHidden2(props) {
  const htmlProps = useVisuallyHidden(props);
  return _MFAH5VLWcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useVisuallyHidden = useVisuallyHidden; exports.VisuallyHidden = VisuallyHidden;
