// @ariakit/react-core/cjs/__chunks/E3JIN4WH.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _EPHIWWY3cjs = require('./EPHIWWY3.cjs');


var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');

// src/dialog/dialog-context.tsx
var _react = require('react');
var ctx = _MFAH5VLWcjs.createStoreContext.call(void 0, 
  [_EPHIWWY3cjs.DisclosureContextProvider],
  [_EPHIWWY3cjs.DisclosureScopedContextProvider]
);
var useDialogContext = ctx.useContext;
var useDialogScopedContext = ctx.useScopedContext;
var useDialogProviderContext = ctx.useProviderContext;
var DialogContextProvider = ctx.ContextProvider;
var DialogScopedContextProvider = ctx.ScopedContextProvider;
var DialogHeadingContext = _react.createContext.call(void 0, void 0);
var DialogDescriptionContext = _react.createContext.call(void 0, void 0);









exports.useDialogContext = useDialogContext; exports.useDialogScopedContext = useDialogScopedContext; exports.useDialogProviderContext = useDialogProviderContext; exports.DialogContextProvider = DialogContextProvider; exports.DialogScopedContextProvider = DialogScopedContextProvider; exports.DialogHeadingContext = DialogHeadingContext; exports.DialogDescriptionContext = DialogDescriptionContext;
