// @ariakit/react-core/cjs/__chunks/FKDBPJ3J.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});





var _H3K3UH33cjs = require('./H3K3UH33.cjs');



var _2P3WACMLcjs = require('./2P3WACML.cjs');



var _RREOTJCIcjs = require('./RREOTJCI.cjs');


var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');

// src/menu/menu-context.tsx
var _react = require('react');
var menu = _MFAH5VLWcjs.createStoreContext.call(void 0, 
  [_2P3WACMLcjs.CompositeContextProvider, _RREOTJCIcjs.HovercardContextProvider],
  [_2P3WACMLcjs.CompositeScopedContextProvider, _RREOTJCIcjs.HovercardScopedContextProvider]
);
var useMenuContext = menu.useContext;
var useMenuScopedContext = menu.useScopedContext;
var useMenuProviderContext = menu.useProviderContext;
var MenuContextProvider = menu.ContextProvider;
var MenuScopedContextProvider = menu.ScopedContextProvider;
var useMenuBarContext = _H3K3UH33cjs.useMenubarContext;
var useMenuBarScopedContext = _H3K3UH33cjs.useMenubarScopedContext;
var useMenuBarProviderContext = _H3K3UH33cjs.useMenubarProviderContext;
var MenuBarContextProvider = _H3K3UH33cjs.MenubarContextProvider;
var MenuBarScopedContextProvider = _H3K3UH33cjs.MenubarScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);













exports.useMenuContext = useMenuContext; exports.useMenuScopedContext = useMenuScopedContext; exports.useMenuProviderContext = useMenuProviderContext; exports.MenuContextProvider = MenuContextProvider; exports.MenuScopedContextProvider = MenuScopedContextProvider; exports.useMenuBarContext = useMenuBarContext; exports.useMenuBarScopedContext = useMenuBarScopedContext; exports.useMenuBarProviderContext = useMenuBarProviderContext; exports.MenuBarContextProvider = MenuBarContextProvider; exports.MenuBarScopedContextProvider = MenuBarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
