// @ariakit/react-core/cjs/__chunks/H3K3UH33.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _2P3WACMLcjs = require('./2P3WACML.cjs');


var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');

// src/menubar/menubar-context.tsx
var _react = require('react');
var menubar = _MFAH5VLWcjs.createStoreContext.call(void 0, 
  [_2P3WACMLcjs.CompositeContextProvider],
  [_2P3WACMLcjs.CompositeScopedContextProvider]
);
var useMenubarContext = menubar.useContext;
var useMenubarScopedContext = menubar.useScopedContext;
var useMenubarProviderContext = menubar.useProviderContext;
var MenubarContextProvider = menubar.ContextProvider;
var MenubarScopedContextProvider = menubar.ScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);








exports.useMenubarContext = useMenubarContext; exports.useMenubarScopedContext = useMenubarScopedContext; exports.useMenubarProviderContext = useMenubarProviderContext; exports.MenubarContextProvider = MenubarContextProvider; exports.MenubarScopedContextProvider = MenubarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
