// @ariakit/react-core/cjs/__chunks/P66OGEAY.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _2P3WACMLcjs = require('./2P3WACML.cjs');



var _5X5VM34Kcjs = require('./5X5VM34K.cjs');


var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');

// src/select/select-context.tsx
var _react = require('react');
var ctx = _MFAH5VLWcjs.createStoreContext.call(void 0, 
  [_5X5VM34Kcjs.PopoverContextProvider, _2P3WACMLcjs.CompositeContextProvider],
  [_5X5VM34Kcjs.PopoverScopedContextProvider, _2P3WACMLcjs.CompositeScopedContextProvider]
);
var useSelectContext = ctx.useContext;
var useSelectScopedContext = ctx.useScopedContext;
var useSelectProviderContext = ctx.useProviderContext;
var SelectContextProvider = ctx.ContextProvider;
var SelectScopedContextProvider = ctx.ScopedContextProvider;
var SelectItemCheckedContext = _react.createContext.call(void 0, false);
var SelectHeadingContext = _react.createContext.call(void 0, null);









exports.useSelectContext = useSelectContext; exports.useSelectScopedContext = useSelectScopedContext; exports.useSelectProviderContext = useSelectProviderContext; exports.SelectContextProvider = SelectContextProvider; exports.SelectScopedContextProvider = SelectScopedContextProvider; exports.SelectItemCheckedContext = SelectItemCheckedContext; exports.SelectHeadingContext = SelectHeadingContext;
