// @ariakit/react-core/cjs/__chunks/XCCLPY67.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _XHEEGBWUcjs = require('./XHEEGBWU.cjs');


var _E3JIN4WHcjs = require('./E3JIN4WH.cjs');




var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');


var _TM5Y6AJUcjs = require('./TM5Y6AJU.cjs');




var _7EQBAZ46cjs = require('./7EQBAZ46.cjs');

// src/dialog/dialog-dismiss.tsx
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
var TagName = "button";
var useDialogDismiss = _MFAH5VLWcjs.createHook.call(void 0, 
  function useDialogDismiss2(_a) {
    var _b = _a, { store } = _b, props = _7EQBAZ46cjs.__objRest.call(void 0, _b, ["store"]);
    const context = _E3JIN4WHcjs.useDialogScopedContext.call(void 0, );
    store = store || context;
    const onClickProp = props.onClick;
    const onClick = _TM5Y6AJUcjs.useEvent.call(void 0, (event) => {
      onClickProp == null ? void 0 : onClickProp(event);
      if (event.defaultPrevented) return;
      store == null ? void 0 : store.hide();
    });
    const children = _react.useMemo.call(void 0, 
      () => /* @__PURE__ */ _jsxruntime.jsxs.call(void 0, 
        "svg",
        {
          "aria-label": "Dismiss popup",
          display: "block",
          fill: "none",
          stroke: "currentColor",
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: 1.5,
          viewBox: "0 0 16 16",
          height: "1em",
          width: "1em",
          children: [
            /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "line", { x1: "4", y1: "4", x2: "12", y2: "12" }),
            /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "line", { x1: "4", y1: "12", x2: "12", y2: "4" })
          ]
        }
      ),
      []
    );
    props = _7EQBAZ46cjs.__spreadProps.call(void 0, _7EQBAZ46cjs.__spreadValues.call(void 0, {
      "data-dialog-dismiss": "",
      children
    }, props), {
      onClick
    });
    props = _XHEEGBWUcjs.useButton.call(void 0, props);
    return props;
  }
);
var DialogDismiss = _MFAH5VLWcjs.forwardRef.call(void 0, function DialogDismiss2(props) {
  const htmlProps = useDialogDismiss(props);
  return _MFAH5VLWcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useDialogDismiss = useDialogDismiss; exports.DialogDismiss = DialogDismiss;
