// @ariakit/react-core/cjs/__chunks/XHEEGBWU.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _BBHOBWVOcjs = require('./BBHOBWVO.cjs');




var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');



var _TM5Y6AJUcjs = require('./TM5Y6AJU.cjs');



var _7EQBAZ46cjs = require('./7EQBAZ46.cjs');

// src/button/button.tsx
var _dom = require('@ariakit/core/utils/dom');
var _react = require('react');
var TagName = "button";
var useButton = _MFAH5VLWcjs.createHook.call(void 0, 
  function useButton2(props) {
    const ref = _react.useRef.call(void 0, null);
    const tagName = _TM5Y6AJUcjs.useTagName.call(void 0, ref, TagName);
    const [isNativeButton, setIsNativeButton] = _react.useState.call(void 0, 
      () => !!tagName && _dom.isButton.call(void 0, { tagName, type: props.type })
    );
    _react.useEffect.call(void 0, () => {
      if (!ref.current) return;
      setIsNativeButton(_dom.isButton.call(void 0, ref.current));
    }, []);
    props = _7EQBAZ46cjs.__spreadProps.call(void 0, _7EQBAZ46cjs.__spreadValues.call(void 0, {
      role: !isNativeButton && tagName !== "a" ? "button" : void 0
    }, props), {
      ref: _TM5Y6AJUcjs.useMergeRefs.call(void 0, ref, props.ref)
    });
    props = _BBHOBWVOcjs.useCommand.call(void 0, props);
    return props;
  }
);
var Button = _MFAH5VLWcjs.forwardRef.call(void 0, function Button2(props) {
  const htmlProps = useButton(props);
  return _MFAH5VLWcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useButton = useButton; exports.Button = Button;
