// @ariakit/react-core/cjs/__chunks/XQCGTEP2.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _CSAYKKWDcjs = require('./CSAYKKWD.cjs');




var _MFAH5VLWcjs = require('./MFAH5VLW.cjs');



var _7EQBAZ46cjs = require('./7EQBAZ46.cjs');

// src/focus-trap/focus-trap.tsx
var TagName = "span";
var useFocusTrap = _MFAH5VLWcjs.createHook.call(void 0, 
  function useFocusTrap2(props) {
    props = _7EQBAZ46cjs.__spreadProps.call(void 0, _7EQBAZ46cjs.__spreadValues.call(void 0, {
      "data-focus-trap": "",
      tabIndex: 0,
      "aria-hidden": true
    }, props), {
      style: _7EQBAZ46cjs.__spreadValues.call(void 0, {
        // Prevents unintended scroll jumps.
        position: "fixed",
        top: 0,
        left: 0
      }, props.style)
    });
    props = _CSAYKKWDcjs.useVisuallyHidden.call(void 0, props);
    return props;
  }
);
var FocusTrap = _MFAH5VLWcjs.forwardRef.call(void 0, function FocusTrap2(props) {
  const htmlProps = useFocusTrap(props);
  return _MFAH5VLWcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useFocusTrap = useFocusTrap; exports.FocusTrap = FocusTrap;
